common/app/model/Badges.scala

```diff
@@ -17,6 +17,9 @@ object Badges {
     else Seq()
   }
 
-  def badgeFor(c: ContentType) = allBadges.find(badge => c.tags.tags.exists(tag => tag.id == badge.seriesTag))
+  def badgeFor(c: ContentType) = c.tags.tags.map(_.id).foldLeft(None: Option[Badge]) { (maybeBadge, tag) =>
+      maybeBadge orElse allBadges.find(b => b.seriesTag == tag)
```

Comment:
clever way of doing it!  now you'll loop through the whole list of tags even if you already found a badge, but it will be a short list so not a problem